_posts/yiyangshi/2016-04-06-YiyangDictionaryExercise.md

```diff
@@ -0,0 +1,14 @@
+---
+author: yiyangshi
+layout: post
+title: "Yiyang's Dictionary Exercise"
+---
+
+Exercise 1:
+<iframe src="https://trinket.io/embed/python3/7ce0bf91d6" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Exercise 2 to 5:
+<iframe src="https://trinket.io/embed/python3/aeff4e198d" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Reflection:
+I think the exercises really helped me understand the different applications of dictionary. It can be used for counting frequence, finding specific informations, finding maximum, etc. I am especially surprised by how easy it is to use the dictionary to find the email address that sends email the most. I am still a little rust on special characters. I guess I just need more exercises in order to remember all the special characters. 
```

Comment:
Nice!

Don't feel the need to memorize regex codes.  Just use a reference and the ones you use often you'll remember automatically.